Mathlib/Algebra/Order/Field/Rat.lean

```diff
@@ -32,3 +32,7 @@ end Rat
 -- instances for performance
 deriving instance CanonicallyLinearOrderedSemifield, LinearOrderedSemifield,
   LinearOrderedCommGroupWithZero for NNRat
+
+/-- These test cases ensure there is no diamond with the `Semifield` structure defined earlier. -/
+example : (inferInstanceAs (Semifield ℚ≥0)).toCommGroupWithZero =
+    (inferInstanceAs (LinearOrderedCommGroupWithZero ℚ≥0)).toCommGroupWithZero := rfl
```

Comment:
Maybe this belongs in `tests/instance_diamonds.lean`?

Reply:
Sounds good! I need to go for a moment, would you like to do the honours? :)